lib/matplotlib/tests/test_ticker.py

```diff
@@ -524,26 +524,93 @@ def test_basic(self, format, input, expected):
 
 
 class TestEngFormatter(object):
-    format_data = [
-        ('', 0.1, u'100 m'),
-        ('', 1, u'1'),
-        ('', 999.9, u'999.9'),
-        ('', 1001, u'1.001 k'),
-        (u's', 0.1, u'100 ms'),
-        (u's', 1, u'1 s'),
-        (u's', 999.9, u'999.9 s'),
-        (u's', 1001, u'1.001 ks'),
+    # (input, expected) where ''expected'' corresponds to the outputs
+    # respectively returned when (places=None, places=0, places=2)
+    raw_format_data = [
+        (-1234.56789, ('-1.23457 k', '-1 k', '-1.23 k')),
+        (-1.23456789, ('-1.23457', '-1', '-1.23')),
+        (-0.123456789, ('-123.457 m', '-123 m', '-123.46 m')),
+        (-0.00123456789, ('-1.23457 m', '-1 m', '-1.23 m')),
+        (-0.0, ('0', '0', '0.00')),
+        (-0, ('0', '0', '0.00')),
+        (0, ('0', '0', '0.00')),
+        (1.23456789e-6, ('1.23457 \u03bc', '1 \u03bc', '1.23 \u03bc')),
+        (0.123456789, ('123.457 m', '123 m', '123.46 m')),
+        (0.1, ('100 m', '100 m', '100.00 m')),
+        (1, ('1', '1', '1.00')),
+        (1.23456789, ('1.23457', '1', '1.23')),
+        (999.9, ('999.9', '999', '999.90')),
+        (1000, ('1 k', '1 k', '1.00 k')),
+        (1001, ('1.001 k', '1 k', '1.00 k')),
+        (100001, ('100.001 k', '100 k', '100.00 k')),
+        (987654.321, ('987.654 k', '987 k', '987.65 k'))
     ]
 
-    @pytest.mark.parametrize('unit, input, expected', format_data)
-    def test_formatting(self, unit, input, expected):
+    @pytest.mark.parametrize('input, expected', raw_format_data)
+    def test_params(self, input, expected):
         """
-        Test the formatting of EngFormatter with some inputs, against
-        instances with and without units. Cases focus on when no SI
-        prefix is present, for values in [1, 1000).
+        Test the formatting of EngFormatter for various values of the 'places'
+        argument, in several cases:
+            0.  without a unit symbol but with a (default) space separator;
+            1.  with both a unit symbol and a (default) space separator;
+            2x. with both a unit symbol and some non default separators;
```

Comment:
I was using it to indicate that both cases 2. and 3. were looped over several subcases ("2a", "2b", etc. in my head…). But the meaning is not obvious, so I will just switch for "2." and "3."